lib/net/ssh/proxy/command.rb

```diff
@@ -93,7 +93,7 @@ def io.recv(size)
             result = read_nonblock(size)
           rescue IO::WaitReadable, Errno::EINTR
             timeout_in_seconds = 20
-            if IO.select([self], nil, [self], timeout_in_seconds) == nil
+            if wait_readable(timeout_in_seconds) == nil
```

Comment:
It does return nil on timeout. Documentation is not very clear, but experiment it in irb and you'll see.